contracts/BackedOracle.sol

```diff
@@ -0,0 +1,154 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+interface AggregatorInterface {
+  function latestAnswer() external view returns (int256);
+
+  function latestTimestamp() external view returns (uint256);
+
+  function latestRound() external view returns (uint256);
+
+  function getAnswer(uint256 roundId) external view returns (int256);
+
+  function getTimestamp(uint256 roundId) external view returns (uint256);
+
+  event AnswerUpdated(int256 indexed current, uint256 indexed roundId, uint256 updatedAt);
+  event NewRound(uint256 indexed roundId, address indexed startedBy, uint256 startedAt);
+}
+
+interface AggregatorV3Interface {
+  function decimals() external view returns (uint8);
+  function description() external view returns (string memory);
+
+  // getRoundData and latestRoundData should both raise "No data present"
+  // if they do not have data to report, instead of returning unset values
+  // which could be misinterpreted as actual reported values.
+  function getRoundData(uint80 _roundId)
+  external
+  view
+  returns (
+    uint80 roundId,
+    int256 answer,
+    uint256 startedAt,
+    uint256 updatedAt,
+    uint80 answeredInRound
+  );
+
+  function latestRoundData()
+  external
+  view
+  returns (
+    uint80 roundId,
+    int256 answer,
+    uint256 startedAt,
+    uint256 updatedAt,
+    uint80 answeredInRound
+  );
+
+}
+
+interface AggregatorV2V3Interface is AggregatorInterface, AggregatorV3Interface
+{
+}
+
+contract BackedOracle is AggregatorV2V3Interface, Ownable {
+  struct RoundData {
+    int192 answer;
+    uint32 timestamp;
+  }
+
+  uint8 private _decimals;
+  string private _description;
+
+  mapping(uint256 => RoundData) private _roundData;
+  uint80 private _latestRoundNumber;
+
+  constructor(uint8 __decimals, string memory __description) {
+    _decimals = __decimals;
+    _description = __description;
+  }
+
+  function decimals() external view override returns (uint8) {
+    return _decimals;
+  }
+
+  function description() external view override returns (string memory) {
+    return _description;
+  }
+
+  function latestAnswer() external view override returns (int256) {
```

Comment:
latestAnswer, latestTimestamp, latestRound, getAnswer, and getTimestamp should revert if round doesn't exist, instead they return the 0 value, which could be confusing. The comment in line 47 mentions that getRoundData and latestRoundData should revert in that case for this exact reason, and I think this applies the same to the methods listed here.

Reply:
this is done in dc683b0ec7df2a91033d121e24e0142d8da2a476 except for getAnswer and getRoundData - these check if the `answer != 0`. In theory it is possible that the price of an assert goes to 0 (even if with the fact that the price can move by at most 10% an hour it might take some time before the admin can actually set the price to 0)

Reply:
I think it might be better to just add a require check that the newAnswer is not 0 in the updateAnswer function.